### PR TITLE
Add Oracle Cloud DevOps CI Workflow

### PR DESCRIPTION
**Pull request by n0tl3ss**
This PR adds `Oracle Cloud DevOps Build CI Workflow` feature. 

- It adds `build_spec.yml` into the root of the generated project that is required for the Oracle Cloud DevOps Build tool.
- Adds test that verify generated file

Docs that are related: 
- https://docs.oracle.com/en-us/iaas/Content/devops/using/build_specs.htm - file format specification
- https://docs.oracle.com/en-us/iaas/Content/devops/using/devops_iampolicies.htm - how to setup CI

### CODE REVIEW

**Review by alvarosanchez — APPROVED** [review]

**Review by wetted — APPROVED** [review]

**Comment by graemerocher** [comment]
Needs a few improvements:

1. if the GraalVM feature is selected we should build a native image
2. We need a way to generate a build that builds docker images (both native and not native) since the output artefact can be a `DOCKER_IMAGE`. See https://docs.oracle.com/en-us/iaas/Content/devops/using/build_specs.htm

**Review by graemerocher — APPROVED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/oci/OCICiWorkflowFeature.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import io.micronaut.starter.template.RockerTemplate;
+import io.micronaut.starter.template.Template;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class OCICiWorkflowFeature extends CIWorkflowFeature {
+
+    public static final String NAME = "oci-devops-build-ci";
+    private static final String WORKFLOW_FILENAME = "build_spec.yml";
+    private static final String GRAALVM_VERSION = "22.1.0";
```
Comment (wetted) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/oci/OCICiWorkflowFeature.java]:
```suggestion
```

File: starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/oci/OCICiWorkflowFeature.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    generatorContext.getProject().getName(),
+                    generatorContext.getJdkVersion(),
+                    generatorContext.getBuildTool()
+            ));
+        } else {
+            return new RockerTemplate(getWorkflowFileName(generatorContext), buildSpecGraal.template(
+                    generatorContext.getProject().getName(),
+                    generatorContext.getJdkVersion(),
+                    generatorContext.getBuildTool(),
+                    GRAALVM_VERSION
```
Comment (wetted) [inline starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/oci/OCICiWorkflowFeature.java]:
```suggestion
                    VersionInfo.getDependencyVersion("graal").value()
```

**Review by wetted — CHANGES_REQUESTED** [review]
Avoid hard coding versions. Use `VersionInfo` API whenever possible

**Review by wetted — APPROVED** [review]

**Pull request merged**